rust/operator-binary/src/backend/tls.rs

```diff
@@ -36,6 +38,23 @@ use super::{
     SecretBackend, SecretBackendError, SecretContents,
 };
 
+/// As the Pods will be evicted [`EXPIRE_PODS_BEFORE_CERT_EXPIRES_PERIOD`] before
+/// the cert actually expires, this results in a restart in approx every 2 weeks,
+/// which matches the rolling re-deploy of k8s nodes of e.g.:
+/// * 1 week for IONOS
+/// * 2 weeks for some on-prem k8s clusters
+pub const DEFAULT_MAX_CERT_LIFETIME: std::time::Duration =
+    std::time::Duration::from_secs(15 * 24 * 60 * 60);
+/// This needs to be sufficiently larger than [`EXPIRE_PODS_BEFORE_CERT_EXPIRES_PERIOD`], so that the
+/// Pod does not immediately gets evicted once it's created.
+pub const MIN_MAX_CERT_LIFETIME: Duration = Duration::days(1);
+
+/// When a StatefulSet has many Pods (e.g. 80 HDFS datanodes or Trino workers) a rolling
+/// redeployment can take multiple hours. When the certificates of all datanodes expire approximately
+/// at the same time and PodDisruptionBudgets are in place, Pods can need to this time to properly shut down,
+/// so we need to evict them enough time in advance
+pub const EXPIRE_PODS_BEFORE_CERT_EXPIRES_PERIOD: Duration = Duration::hours(6);
```

Comment:
This should probably be calculated as some fraction of the cert lifetime, and/or be configurable as well.

Reply:
I was thinking about a percentage at first as well, but I think the time it takes for a product to gracefully shut down is mostly the same, regardless of the uptime so far.
It is now also configurable via a Volume annotation

Reply:
Implementation switched to what we have decided upon in the Arch meeting